timemachine/fe/mcgregor.py

```diff
@@ -243,39 +243,48 @@ def mcs(
     enforce_core_core,
     connected_core,
     min_threshold,
+    initial_mapping,
     filter_fxn: Callable[[Sequence[int]], bool] = lambda core: True,
 ) -> Tuple[List[NDArray], List[NDArray], MCSDiagnostics]:
     assert n_a <= n_b
 
+    predicate = build_predicate_matrix(n_a, n_b, priority_idxs)
     g_a = Graph(n_a, bonds_a)
     g_b = Graph(n_b, bonds_b)
+    base_marcs = _initialize_marcs_given_predicate(g_a, g_b, predicate)
 
-    predicate = build_predicate_matrix(n_a, n_b, priority_idxs)
-    marcs = _initialize_marcs_given_predicate(g_a, g_b, predicate)
+    base_map_a_to_b = [UNMAPPED] * n_a
+    base_map_b_to_a = [UNMAPPED] * n_b
+    if initial_mapping is not None:
+        for a, b in initial_mapping:
+            base_map_a_to_b[a] = b
+            base_map_b_to_a[b] = a
+            base_marcs = refine_marcs(g_a, g_b, a, b, base_marcs)
 
+    base_layer = len(initial_mapping)
     priority_idxs = tuple(tuple(x) for x in priority_idxs)
     # Keep start time for debugging purposes below
     start_time = time.time()  # noqa
 
     mcs_result = None
 
     # run in reverse by guessing max # of edges to avoid getting stuck in minima.
-    max_threshold = _arcs_left(marcs)
+    max_threshold = _arcs_left(base_marcs)
     total_nodes_visited = 0
     for idx in range(max_threshold):
         cur_threshold = max_threshold - idx
         if cur_threshold < min_threshold:
             raise NoMappingError(f"Unable to find mapping with at least {min_threshold} edges")
-        map_a_to_b = [UNMAPPED] * n_a
-        map_b_to_a = [UNMAPPED] * n_b
+        map_a_to_b = copy.deepcopy(base_map_a_to_b)
+        map_b_to_a = copy.deepcopy(base_map_b_to_a)
```

Comment:
I was surprised that I could remove the `copy.deepcopy` from these lines and the tests all still passed. Does this indicate useful additional test cases, or that the copy is somehow not needed?

Reply:
Also, I don't think `deepcopy` should be needed here, since I think these are lists of `int`s?

Reply:
Per your comment about the being able to remove the copy not resulting in any copy, it is because the final map returned is the initial `base_map_a_to_b`. After we update the index, we pop it. https://github.com/proteneer/timemachine/blob/master/timemachine/fe/mcgregor.py#L421-L446

Verified this by adding in an assertion that the `map_a_to_b` at the end of recursion is identical to `base_map_a_to_b` and b_to_a. 

I would probably advocate for keeping the copy just to be super safe, but does not appear strictly neccessary